cpukit/score/cpu/arm/cpu_asm.S

```diff
@@ -65,7 +65,10 @@ DEFINE_FUNCTION_ARM(_CPU_Context_switch)
 #endif
 
 	str	r3, [r0, #ARM_CONTEXT_CONTROL_ISR_DISPATCH_DISABLE]
-
+ // #if defined ( USE_THREAD_STACK_PROTECTION )
+	str r0, [r0, #44]
+	bl  prot_stack_context_switch
```

Comment:
How are you passing the argument to this function?

Reply:
This is a mistake which I have fixed on my local branch, the instruction should be `ldr r0, [r0, #ARM_STACK_PROT_ATTR_OFFSET]`. This way the argument will be passed through the r0 register.

Reply:
But even after your fix, the code won’t behave correctly after the function returns as you’ve overwritten ‘r0’

Reply:
Oh, sorry for the confusion. The complete code is
`mov r1, r0`
`ldr r0, [r0,  #ARM_STACK_PROT_ATTR_OFFSET]`
`bl prot_stack_context_switch`
`mov r0, r1`

Reply:
No problem, thanks for posting the code